ast/misc.go

```diff
@@ -2035,16 +2037,40 @@ type TableOptimizerHint struct {
 	HintFlag         bool
 }
 
+// HintTable is table in the hint. It may have query block info.
+type HintTable struct {
+	TableName model.CIStr
+	QBName    model.CIStr
```

Comment:
Why need we save the `QBName` in both `HintTable` and `TableOptimizerHint`?

Reply:
Because we can specify the QB within which the hint will take effect, and also specify a table in some QB?

Reply:
For example: `/*+ HASH_JOIN(@sel1 t1, t2) */` and `/*+ HASH_JOIN(t1@sel1, t2@sel2)`.

Reply:
Yes, you can take a look at the test.